src/main/attached_database.cpp

```diff
@@ -54,6 +77,14 @@ AttachedDatabase::AttachedDatabase(DatabaseInstance &db, Catalog &catalog_p, Sto
 }
 
 AttachedDatabase::~AttachedDatabase() {
+
+	if (!IsSystem() && !catalog->InMemory()) {
+		auto &db_manager = db.GetDatabaseManager();
+		lock_guard<mutex> write_lock(db.GetDatabaseManager().GetDbPathsLock());
+		auto &db_paths = db_manager.GetDbPaths();
+		db_paths.erase(db_paths.find(storage->GetDBPath()));
```

Comment:
Can we also add a test where we attach -> detach -> attach the same file to ensure the path is correctly erased? (perhaps that test already exists in which case it's fine to not add another one)